pkgs/applications/audio/soundconverter/default.nix

```diff
@@ -0,0 +1,55 @@
+{ pkgs, python3Packages, lib, fetchurl }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.3";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "sha256-hzIG/4LD3705erPYvXb7uoRwF9LtKKIKB3jrhpYMsZ0=";
+  };
+
+  # FIXME: Tests still fail with:
+  # 'No GSettings schemas are installed on the system'
```

Comment:
You might need to set `XDG_DATA_DIRS` in `preCheck` like here:

https://github.com/NixOS/nixpkgs/blob/3910e719cecc511d9e6215a61a51df69fe1a218c/pkgs/applications/networking/calls/default.nix#L88

Reply:
Right, I tried that with:
```nix
{
  preCheck = let
    gst = pkgs.gsettings-desktop-schemas;
  in ''
    export XDG_DATA_DIRS=${gst}/share/gsettings-schemas/${gst.name}:$XDG_DATA_DIRS
  '';
}
```
But then I just get:
```
(Lib-GIO-ERROR **: 19:12:50.649: Settings schema 'org.soundconverter' is not installed
```

Reply:
I managed to get a bit forward with tests using this:
```nix
{
  preCheck = let
    gst = pkgs.gsettings-desktop-schemas;
  in ''
    addToSearchPath XDG_DATA_DIRS "${gst}/share/gsettings-schemas/${gst.name}"
    addToSearchPath XDG_DATA_DIRS "$out/share/gsettings-schemas/${pname}-${version}"
  '';
}
```
But now I'm getting these:
```
ERROR: Disabling flac-enc output. Do you have "gst-plugins-good" installed?
ERROR: Disabling wav-enc output. Do you have "gst-plugins-good" installed?
ERROR: Disabling vorbis-enc output. Do you have "gst-plugins-base" installed?
ERROR: Disabling vorbis-mux output. Do you have "gst-plugins-base" installed?
ERROR: Disabling mp3-id-tags output. Do you have "gst-plugins-good" installed?
ERROR: Disabling mp3-vbr-tags output. Do you have "gst-plugins-ugly" installed?
ERROR: Disabling mp3-enc output. Do you have "gst-plugins-good" installed?
ERROR: Disabling aac-mux output. Do you have "gst-plugins-good" installed?
ERROR: Disabling opus-enc output. Do you have "gst-plugins-base" installed?
ERROR: Disabling aac-enc output. Do you have "gst-plugins-bad" installed?
```
Tests seem to be harder to get to work then the app itself.

Reply:
You will need to add the directory containing this packages’s `glib-2.0/schemas/gschemas.compiled`.

Reply:
I managed to fix some errors with this:
```nix
{
  preCheck = let
    gst = pkgs.gsettings-desktop-schemas;
  in ''
    addToSearchPath XDG_DATA_DIRS "${gst}/share/gsettings-schemas/${gst.name}"
    addToSearchPath XDG_DATA_DIRS "$out/share/gsettings-schemas/${pname}-${version}"
    addToSearchPath PYTHONPATH "$PWD/tests"
    export HOME=$(mktemp -d)
  '';
}
```
But I'm still getting this all over the place:
```
WARNING: fdkaacenc is recommended for aac conversion but it is not available. It can be installed with gst-plugins-bad. Using avenc_aac instead.
ERROR: gstreamer error when creating pipeline: gst_parse_error: no element "avenc_aac" (1): /build/soundconverter-4.0.3/tests/test data/audio/b/c.mp3
```
So it's not seeing gstreamer plugins when running tests.

Reply:
Try checking `$GST_PLUGIN_SYSTEM_PATH_1_0` env var.

Reply:
I don't think the tests are gonna fly because they require an X server:
```
Unable to init server: Could not connect: Connection refused
```
So the only sensible thing is to keep them disabled.

Reply:
You  can use xvfb for that: https://github.com/NixOS/nixpkgs/blob/0e57783d3a9ff01d3ace41d9f75d84b865285746/pkgs/development/libraries/libhandy/default.nix#L103-L104

Reply:
I tried using `xvfb-run` and it kinds worked, but it would get stuck forever on:
```
ERROR: gstreamer error when creating pipeline: gst_parse_error: no element "faac" (1): /build/soundconverter-4.0.3/tests/test data/audio/a.wav
```
So I did a fresh Ubuntu install in a VM and tried to run tests there, and the same thing happens:

![image](https://user-images.githubusercontent.com/2212681/147793077-f904f56a-d6d7-422a-a87c-f8943271eba8.png)

I'm not sure this isn't broken overall.

Reply:
We are [not building `faac` by default](https://github.com/NixOS/nixpkgs/blob/1e0394354e7feb1cd9476eb44eb2279a645de370/pkgs/development/libraries/gstreamer/bad/default.nix#L14) and I suspect Ubuntu does not have `fdkaacenc` installed. You will need to use `(gst_all_1.gst-plugins-bad.override { faacSupport = true; })`.

Reply:
I don't think that's gonna work, since `faac` has an unfree license:

https://github.com/NixOS/nixpkgs/blob/77b45a11db3ccbe566bdf3fd397633bd96be466e/pkgs/development/libraries/faac/default.nix#L33

Seems sensible to keep it out by default, since I think it's better to keep the package free. If someone wants to build it with `faac` support they can override the dependency themselves.

So either we disable tests, or we disable the tests that check use of `faac`.

Reply:
I am able to have the tests pass by adding this:
```sh
    substituteInPlace  tests/testcases/integration.py --replace \
      "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
      "for encoder in ['fdkaacenc', 'avenc_aac']:"
```
But I had to add `fdk-aac-encoder` and `gst_all_1.gst-libav` to dependencies.

Reply:
I made it optional with same `faacSupport` boolean.